main/lsp/DefLocSaver.cc

```diff
@@ -53,10 +53,13 @@ ast::ExpressionPtr DefLocSaver::postTransformUnresolvedIdent(core::Context ctx,
     auto &id = ast::cast_tree_nonnull<ast::UnresolvedIdent>(tree);
     if (id.kind == ast::UnresolvedIdent::Kind::Instance || id.kind == ast::UnresolvedIdent::Kind::Class) {
         core::ClassOrModuleRef klass;
-        // Logic cargo culted from `global2Local` in `walker_build.cc`.
         if (id.kind == ast::UnresolvedIdent::Kind::Instance) {
-            ENFORCE(ctx.owner.isMethod());
-            klass = ctx.owner.owner(ctx).asClassOrModuleRef();
+            ENFORCE(ctx.owner.isMethod() || ctx.owner.isClassOrModule());
+            if (ctx.owner.isMethod()) {
+                klass = ctx.owner.owner(ctx).asClassOrModuleRef();
+            } else if (ctx.owner.isClassOrModule()) {
+                klass = ctx.owner.asClassOrModuleRef();
+            }
```

Comment:
This should be equivalent, and I think captures the intent here:

```suggestion
            klass = ctx.owner.enclosingClass(ctx);
```

See https://github.com/sorbet/sorbet/blob/5bdbbe9e/core/Symbols.cc#L2073 for implementation

Reply:
TIL: `enclosingClass`